src/block/RedMushroom.php

```diff
@@ -42,11 +41,18 @@ public function onNearbyBlockChange() : void{
 	}
 
 	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
-		$down = $this->getSide(Facing::DOWN);
-		if(!$down->isTransparent()){
+		if($this->canBeSupportedBy($this->getSide(Facing::DOWN))){
 			return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
 		}
 
 		return false;
 	}
+
+	protected function canBeSupportedBy(Block $block) : bool{
+		$position = $block->getPosition();
+		$lightLevel = $position->getWorld()->getFullLightAt($position->x, $position->y, $position->z);
+		$id = $block->getId();
+		// TODO: nylium support
+		return ($lightLevel <= 12 && !$block->isTransparent()) || $id === BlockLegacyIds::MYCELIUM || $id === BlockLegacyIds::PODZOL;
```

Comment:
The light level here will always be zero if the block is opaque.

Reply:
I'm going to move all this to other pull requests soon if not done before. (Light condition)